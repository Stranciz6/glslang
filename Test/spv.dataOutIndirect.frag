#version 140 

in vec4 Color;

uniform int i;

void main()
{
    gl_FragData[i] = Color;
}
