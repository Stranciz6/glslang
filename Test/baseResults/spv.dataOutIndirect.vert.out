spv.dataOutIndirect.vert
WARNING: 0:3: attribute deprecated in version 130; may be removed in future release
WARNING: 0:4: varying deprecated in version 130; may be removed in future release


Linked vertex stage:


// Module Version 10000
// Generated by (magic number): 80001
// Id's are bound by 39

                              Capability Shader
               1:             ExtInstImport  "GLSL.std.450"
                              MemoryModel Logical GLSL450
                              EntryPoint Vertex 4  "main" 24 27 33 38
                              Source GLSL 130
                              Name 4  "main"
                              Name 8  "i"
                              Name 24  "colorOut"
                              Name 27  "color"
                              Name 33  "gl_Position"
                              Name 38  "gl_VertexID"
                              Decorate 33(gl_Position) BuiltIn Position
                              Decorate 38(gl_VertexID) BuiltIn VertexId
               2:             TypeVoid
               3:             TypeFunction 2
               6:             TypeInt 32 1
               7:             TypePointer Function 6(int)
               9:      6(int) Constant 1
              15:      6(int) Constant 5
              16:             TypeBool
              18:             TypeFloat 32
              19:             TypeVector 18(float) 4
              20:             TypeInt 32 0
              21:     20(int) Constant 6
              22:             TypeArray 19(fvec4) 21
              23:             TypePointer Output 22
    24(colorOut):     23(ptr) Variable Output
              26:             TypePointer Input 19(fvec4)
       27(color):     26(ptr) Variable Input
              29:             TypePointer Output 19(fvec4)
 33(gl_Position):     29(ptr) Variable Output
              34:      6(int) Constant 2
              37:             TypePointer Input 6(int)
 38(gl_VertexID):     37(ptr) Variable Input
         4(main):           2 Function None 3
               5:             Label
            8(i):      7(ptr) Variable Function
                              Store 8(i) 9
                              Branch 10
              10:             Label
              14:      6(int) Load 8(i)
              17:    16(bool) SLessThan 14 15
                              LoopMerge 12 13 None
                              BranchConditional 17 11 12
              11:               Label
              25:      6(int)   Load 8(i)
              28:   19(fvec4)   Load 27(color)
              30:     29(ptr)   AccessChain 24(colorOut) 25
                                Store 30 28
                                Branch 13
              13:               Label
              31:      6(int)   Load 8(i)
              32:      6(int)   IAdd 31 9
                                Store 8(i) 32
                                Branch 10
              12:             Label
              35:     29(ptr) AccessChain 24(colorOut) 34
              36:   19(fvec4) Load 35
                              Store 33(gl_Position) 36
                              Return
                              FunctionEnd
